src/nvim/eval.c

```diff
@@ -22633,14 +22634,15 @@ static inline void term_delayed_free(void **argv)
   term_job_data_decref(j);
 }
 
+/// Called when the terminal buffer gets wiped while the job is still running
 static void term_close(void *d)
 {
   TerminalJobData *data = d;
   if (!data->exited) {
     data->exited = true;
```

Comment:
You're right. In fact, the extra refcount in `f_termopen` should not be required. The terminal doesn't generate callbacks to the job layer after the job died. I removed these bits.